jumble/src/views/CharmDetailView.tsx

```diff
@@ -926,6 +914,20 @@ const DataTab = () => {
     </div>
   );
 
+  let argumentJson: Record<string, any>;
+  try {
+    if (isArgumentExpanded) {
+      argumentJson = translateCellsAndStreamsToPlainJSON(
+        charmManager.getArgument(charm).get(),
+      ) as Record<string, any>;
+    } else {
+      argumentJson = {};
+    }
+  } catch (error) {
+    console.warn("Error translating argument to JSON:", error);
+    argumentJson = {};
+  }
+
```

Comment:
nit: wrapping in `useMemo` would be better again for render perf

Reply:
good idea - should we do it for this below as well:

```
              value={translateCellsAndStreamsToPlainJSON(charm.get()) ?? {}}
```

I had moved it out of inline jsx because getArgument now throws (see above)